src/Interpreters/Context.cpp

```diff
@@ -2029,10 +2029,13 @@ void Context::reloadConfig() const
 
 void Context::shutdown()
 {
-    for (auto & [disk_name, disk] : getDisksMap())
+    if (shared->merge_tree_disk_selector)
```

Comment:
for the record - it was trying to init the disk to shutdown context when an exception was happened before initializing disks. 

```
programs/clickhouse local -q select 1 
Code: 504, e.displayText() = DB::Exception: Unsuccessfull attempt to create working directory: /tmp/clickhouse-local-810185-1603461716-11695049450137462916 exist! (version 20.11.1.1)
terminating with uncaught exception of type DB::Exception: Disk path must ends with '/', but '' doesn't.
Aborted (core dumped)


(gdb) bt
#0  DB::DiskLocal::DiskLocal (this=this@entry=0x7ffff700f218, name_=..., path_=..., keep_free_space_bytes_=0) at ../src/Disks/DiskLocal.h:28
#1  0x00000000092d55be in std::__1::__compressed_pair_elem<DB::DiskLocal, 1, false>::__compressed_pair_elem<char const* const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, int&&, 0ul, 1ul, 2ul> (this=this@entry=0x7ffff700f218, __args=...) at ../contrib/libcxx/include/memory:2214
#2  0x00000000092d3cf5 in std::__1::__compressed_pair<std::__1::allocator<DB::DiskLocal>, DB::DiskLocal>::__compressed_pair<std::__1::allocator<DB::DiskLocal>&, char const* const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, int&&> (this=0x7ffff700f218, __first_args=..., __second_args=..., __pc=...) at ../contrib/libcxx/include/memory:2298
#3  std::__1::__shared_ptr_emplace<DB::DiskLocal, std::__1::allocator<DB::DiskLocal> >::__shared_ptr_emplace<char const* const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int> (this=0x7ffff700f200, __args=@0x7fffffffc7bc: 0, __args=@0x7fffffffc7bc: 0, __args=@0x7fffffffc7bc: 0, __a=...) at ../contrib/libcxx/include/memory:3569
#4  std::__1::make_shared<DB::DiskLocal, char const* const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int> (__args=@0x7fffffffc7bc: 0, 
    __args=@0x7fffffffc7bc: 0, __args=@0x7fffffffc7bc: 0) at ../contrib/libcxx/include/memory:4400
#5  DB::DiskSelector::DiskSelector (this=<optimized out>, config=..., config_prefix=..., context=...) at ../src/Disks/DiskSelector.cpp:43
#6  0x000000000931b9b3 in std::__1::__compressed_pair_elem<DB::DiskSelector, 1, false>::__compressed_pair_elem<Poco::Util::AbstractConfiguration const&, char const* const&, DB::Context const&, 0ul, 1ul, 2ul> (this=this@entry=0x7ffff7004b58, __args=...) at ../contrib/libcxx/include/memory:2214
#7  0x000000000930a449 in std::__1::__compressed_pair<std::__1::allocator<DB::DiskSelector>, DB::DiskSelector>::__compressed_pair<std::__1::allocator<DB::DiskSelector>&, Poco::Util::AbstractConfiguration const&, char const* const&, DB::Context const&> (this=0x7ffff7004b58, __first_args=..., __second_args=..., __pc=...) at ../contrib/libcxx/include/memory:2298
#8  std::__1::__shared_ptr_emplace<DB::DiskSelector, std::__1::allocator<DB::DiskSelector> >::__shared_ptr_emplace<Poco::Util::AbstractConfiguration const&, char const* const&, DB::Context const&> (
    this=0x7ffff7004b40, __args=..., __args=..., __args=..., __a=...) at ../contrib/libcxx/include/memory:3569
#9  std::__1::make_shared<DB::DiskSelector, Poco::Util::AbstractConfiguration const&, char const* const&, DB::Context const&> (__args=..., __args=..., __args=...)
    at ../contrib/libcxx/include/memory:4400
#10 DB::Context::getDiskSelector (this=this@entry=0x7ffff70e2000) at ../src/Interpreters/Context.cpp:1826
#11 0x000000000930a5ff in DB::Context::getDisksMap (this=this@entry=0x7ffff70e2000) at ../src/Interpreters/Context.cpp:1810
#12 0x000000000930c461 in DB::Context::shutdown (this=0x7ffff70e2000) at ../src/Interpreters/Context.cpp:2032
#13 0x000000000482094a in DB::LocalServer::~LocalServer (this=this@entry=0x7fffffffcab8) at ../programs/local/LocalServer.cpp:63
#14 0x0000000004827fe0 in mainEntryClickHouseLocal (argc=<optimized out>, argv=<optimized out>) at ../programs/local/LocalServer.cpp:618
#15 0x0000000004772f13 in main (argc_=<optimized out>, argv_=<optimized out>) at ../programs/main.cpp:400
```

Reply:
```suggestion
    // Disk selector might not be initialized if there was some error during
    // its initialization. Don't try to initialize it again on shutdown.
    if (shared->merge_tree_disk_selector)
```